File: CS/Contacts/Reports/CategoryReportService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraReports.UserDesigner;
using DevExpress.XtraReports.UI;

namespace Contacts.Reports
{
    public class CategoryReportService : ReportTypeService
    {
        private static int _counter = 0;
        public XtraReport GetDefaultReport()
        {
            _counter++;
            CategoryReport r = new CategoryReport();
            r.Name = string.Format("CategoryReport{0}", _counter);
            r.DisplayName = r.Name;
            r.LoadDataSource();
            return r;
        }

        public Type GetType(Type reportType)
        {
            return reportType;
        }
    }
}

File: VB/Contacts/Reports/CategoryReportService.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraReports.UserDesigner
Imports DevExpress.XtraReports.UI

Namespace Contacts.Reports
	Public Class CategoryReportService
        Implements ReportTypeService

        Private Shared _counter As Integer = 0
        Public Function GetDefaultReport() As XtraReport Implements DevExpress.XtraReports.UserDesigner.ReportTypeService.GetDefaultReport
            _counter += 1
            Dim r As New CategoryReport()
            r.Name = String.Format("CategoryReport{0}", _counter)
            r.DisplayName = r.Name
            r.LoadDataSource()
            Return r
        End Function

        Public Function [GetType](ByVal reportType As Type) As Type Implements ReportTypeService.GetType
            Return reportType
        End Function
    End Class
End Namespace
